### DISCUSSION TITLE
Welcome to Helgobox Discussions!

### DISCUSSION THREAD

**Discussion by helgoboss** [helgoboss/helgobox#288] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of the Helgobox community. We hope that you:

  * Ask or answer questions in category [Q&A](https://github.com/helgoboss/helgobox/discussions/categories/q-a)
  * Request new features/improvements or vote for existing ones in category [Ideas](https://github.com/helgoboss/helgobox/discussions/categories/ideas?discussions_q=category%3AIdeas+is%3Aopen+sort%3Adate_created+)
  * Engage with other community members about general stuff in category [General](https://github.com/helgoboss/helgobox/discussions/categories/general)
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

Initially, this "Welcome" post was open for people introducing themselves, which is nice. But that also came with the issue that people would accidentally ask questions in this post, therefore I locked it. Please start a new thread for each question in the category [Q&A](https://github.com/helgoboss/helgobox/discussions/categories/q-a). Thanks for your understanding!

**Comment by vonglan** [comment]
Hello, I love synthesizers and have a software development background. I also love to have a nearly-perfect set of "tools" available (even if my musical output is nothing worth mentioning).
I am enthusiastic about ReaLearn, because it can help with controlling different VST synthesizers in a standard way from different Hardware controllers (especially hardware synthesizers used as controllers), see #205 .
Two years ago, I programmed/configured a couple of JSFX scripts to control various softsynths from the OB6. ReaLearn is the superior  tool for this (except for #203 and, less importantly, #204 , but I hope these will be implemented one day):

- Much simpler implementation/configuration of a mapping
- "relative mode" also possible for VSTs that do not support this in their own "MIDI learn" function
- Feedback if you use encoders with LED rings (like Midifighter Twister or X-Touch Mini)
- With the new approach to decouple controller and VST, it will be relatively easy to add controllers and VSTs

**Comment by jackmau** [comment]
Just discovered github discussion section.

Hi Everyone, I am an Italian jazz-trained pianist/composer by night and an actuary/data scientist by day. I have a long history of looking of way to make electronic instruments more alive through MIDI (and now OSC) control (I am also beta-tester of [MidiPaw](http://www.midipaw.com/)). I have been using realearn since v1, in pair with my purpose built JSFX and I am an avid user of @helgoboss Playtime as well. I think @helgoboss work is outstanding and his Reaper extensions are one of the major reasons Reaper is still the DAW of choice. I particularly ❤️ @helgoboss for having implemented most of my FRs, in particular #6, #11, #12, #160, #186, #469

**Reply by vonglan** [reply]
Hi Jack, interesting.

> Hi Everyone, I am an Italian jazz-trained pianist/composer by day and an actuary/data scientist by day.

You must have a longer day than me ;-)

**Reply by jackmau** [reply]
well it turns out there is no night here ;)

**Comment by tompad2** [comment]
Hi!
My name is Thomas and I come from Sweden. I used to play bass, piano and even tried banjo but got injured in my right hand, so now its all computer and Reaper for doing music. I have been using both CSI and ReaLearn a little for a while. But now its time to choose one of them, and my choice is ReaLearn. Want to use ReaLearn to minimize the use of mouse when working in Reaper. Looking forward to dip my feets in this deep thing :-)

**Comment by Musicicity** [comment]
hi, I have set up realearn with 3 midi controllers so far and it all works with no problem.  I have thoroughly read the documentation but there is one thing I still cannot understand.  what is the difference and reason for the two types of present, main and controller?  they both seems to do the same thing?  what am I missing?  soory for the very basic question.

we learn something every day :)

**Reply by helgoboss** [reply]
Controller compartment allows virtual targets. Main compartment allows virtual sources. Then you can connect those. That's the main difference. Benefits of doing this are described here: https://github.com/helgoboss/realearn/blob/master/doc/user-guide.adoc#controller-compartment

**Reply by Musicicity** [reply]
many thanks for the swift reply, I think I'm beginning to understand...

**Comment by Musicicity** [comment]
I have another question.  Having searched for hours through the documentation I still cannot find the answer.

 

I have an Xtouch and have downloaded the config files for it and mute,solo, rec arm, faders all work well.

 

However I wanted to map other buttons.  I have mapped ‘save’ to save project and this works too.

The question is: Is there a way to light up the ‘save’ button when the project has been modified and turn it off when the project has been saved?

 

Best regards,

 


Michael Norfolk FRSA - Technical Director WaveData Ltd 
+44 7714 322356 mobile
www.wavedata.co.uk <http://www.wavedata.co.uk/>          
P Please consider the environment before printing this e-mail

WaveData Ltd, Concorde House, Comet Way,Eastwood, Essex, England  SS2 6GD United Kingdom    Company Reg.  03998025   VAT No. GB 761 4016 55

**Comment by ElVicente** [comment]
Hi there !
I'm Vincent, I'm a french guitar player and teacher enjoying the exploration of synth music, ( https://on.soundcloud.com/tt5XN if you're curious)
I use several soft and hw stuff (linnstrument !! ♥️) and I'm back to reaper with realearn and playtime 2 for my new live oriented setup.
Thanks again helgoboss for those killer tools =)
